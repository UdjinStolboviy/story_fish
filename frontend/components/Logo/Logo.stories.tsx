import { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { expect } from "@storybook/jest";
import { screen } from "@storybook/testing-library";

import { Logo } from "./Logo";

export default {
  title: "Content/Logo",
  component: Logo,
} as ComponentMeta<typeof Logo>;

export const BasicLogo: ComponentStoryObj<typeof Logo> = {
  play: async () => {
    await expect(screen.getByRole("banner")).toBeInTheDocument();
  },
  args: {
    children: "CoursesBox",
  },
};
