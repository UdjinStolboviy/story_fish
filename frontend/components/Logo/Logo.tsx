import styled from "@emotion/styled";
import { css } from "@emotion/react";

export type Props = {
  /** Logo size in rem */
  size?: number;
};

export const Logo = styled.header<Props>`
  font-family: Monoton;
  font-size: ${({ size = 3 }) => `${size}rem`};
  ${({ theme, size = 3 }) => {
    return css`
      color: ${theme.font.logo};
      text-shadow: 0 0 ${0.1 * size}rem ${theme.font.logoShadow1},
        0 0 ${0.05 * size}rem ${theme.font.logoShadow2},
        0 0 ${0.07 * size}rem ${theme.font.logoShadow2},
        0 0 ${0.08 * size}rem ${theme.font.logoShadow2},
        0 0 ${0.1 * size}rem ${theme.font.logoShadow2};
    `;
  }}
`;
