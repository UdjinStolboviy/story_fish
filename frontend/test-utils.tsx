import { FC, ReactElement } from "react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import { render, RenderOptions } from "@testing-library/react";
import { ThemeProvider } from "@emotion/react";

import { rootReducer, RootState } from "./store";
import { Themes } from "./styles/themes";
import { Layout } from "./components/Layout";

type Options = { preloadedState?: RootState } & RenderOptions;

const customRender = (
  ui: ReactElement,
  { preloadedState, ...options }: Options = {}
) => {
  const store = configureStore({ reducer: rootReducer, preloadedState });

  const Wrapper: FC = ({ children }) => (
    <Provider store={store}>
      <ThemeProvider theme={Themes.light}>{children}</ThemeProvider>
    </Provider>
  );
  return render(ui, { wrapper: Wrapper, ...options });
};

const pageRender = (
  ui: ReactElement,
  { preloadedState, ...options }: Options = {}
) => {
  const store = configureStore({ reducer: rootReducer, preloadedState });

  const Wrapper: FC = ({ children }) => (
    <Provider store={store}>
      <Layout>{children}</Layout>
    </Provider>
  );
  return render(ui, { wrapper: Wrapper, ...options });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, pageRender };
