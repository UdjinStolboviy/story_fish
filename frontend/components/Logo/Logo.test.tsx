import { render } from "@/test-utils";

import { Logo } from "./Logo";

describe("Logo test cases", () => {
  it("Render check", () => {
    const { asFragment } = render(<Logo>CoursesBox</Logo>);
    expect(asFragment()).toMatchSnapshot();
  });
  it("Render with custom size", () => {
    const { asFragment } = render(<Logo size={10}>CoursesBox</Logo>);
    expect(asFragment()).toMatchSnapshot();
  });
});
