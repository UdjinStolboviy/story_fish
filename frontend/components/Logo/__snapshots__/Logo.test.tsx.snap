// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo test cases Render check 1`] = `
<DocumentFragment>
  <header
    class="css-q19ajl"
  >
    CoursesBox
  </header>
</DocumentFragment>
`;

exports[`Logo test cases Render with custom size 1`] = `
<DocumentFragment>
  <header
    class="css-1552zee"
    size="10"
  >
    CoursesBox
  </header>
</DocumentFragment>
`;
